Assets/Kart/kart.fsh

```diff
@@ -32,7 +37,30 @@ out vec4 FragmentColor;
 void main()
 {
     vec3 normal = normalize(v_normal);
-	vec3 light = DirectionalLightDiffuseColor * max(0.0, dot(normal, DirectionalLightDirection));
-    light = clamp(light + AmbientLightColor, 0.0, 1.0);
-    FragmentColor = vec4(light, 1.0) * texture(DiffuseTexture, v_uv1);
+
+    //-----------------------------------------
+    // Directional light computation.
+
+	vec3 DirectionalLight = DirectionalLightDiffuseColor * max(0.0, dot(normal, DirectionalLightDirection));
+
+    //-----------------------------------------
+    // Point light computation.
+
+    vec3 PointLightDirectionVector = PointLightPosition - v_position.xyz;
+    float distance = length(PointLightDirectionVector);
+
+    vec3 PointLight = vec3(0.0, 0.0, 0.0);
+
+    //if (distance <= PointLightRange)
```

Comment:
It's there just to be able to un-comment it if we want a limit on the range of the point light. The PointLightRange parameter is still being passed to use if necessary. Having a range is how we were taught to use point lights but I've never really seen a use for it considering it falls to zero after a certain point in the attenuation calculation anyway and it adds a branch to the fragment shader code.